service/resource/legacy/resource.go

```diff
@@ -251,9 +251,13 @@ func (s *Resource) ApplyUpdateChange(ctx context.Context, obj, updateChange inte
 }
 
 func (s *Resource) processCluster(cluster awstpr.CustomObject) error {
-	// Create cluster namespace in k8s.
-	if err := s.createClusterNamespace(cluster.Spec.Cluster); err != nil {
-		return microerror.Maskf(executionFailedError, fmt.Sprintf("could not create cluster namespace: '%#v'", err))
+	// For new clusters using Cloud Formation there is an OperatorKit resource
+	// for the k8s namespace.
+	if !key.UseCloudFormation(cluster) {
+		// Create cluster namespace in k8s.
+		if err := s.createClusterNamespace(cluster.Spec.Cluster); err != nil {
+			return microerror.Maskf(executionFailedError, fmt.Sprintf("could not create cluster namespace: '%#v'", err))
+		}
```

Comment:
We don't create the namespace in the legacy resource if we're using cloud-formation. 

FYI. Hopefully we can replace `key.UseCloudFormation` with version bundles so we can launch clusters with Cloud Formation on internal installations. I need to look into this.

Reply:
I don't follow this here. Why do we still work on the legacy resource? IMO it should stay legacy as it is and should not be extended anymore. That is why it is legacy. It is deprecated. We should rather extend the new resources and route custom objects accordingly. We can already do this. Check the resource router in operatorkit. We can also talk about it if you like. Defining/exposing version bundles should also be done but this is kind of independent still because we are in the middle of a transition period. We should do it anyway though.

Reply:
I wouldn't say this is extending the legacy resource. It disables creating the namespace if the version is cloud-formation so we use the new resources. This is a transitional state until the new resources provide all the logic to launch a cluster.

Thanks I'll look again at the resource router. Talking it over would also be good. I'll ping you tomorrow.

We discussed in planning that we'd like aws-operator to support 2 versions `legacy` and the new `cloud formation` based operator. We can then expose the CF version internally for testing.

Once the CF version can fully launch clusters we can expose it to customers, deprecate legacy and finally remove legacy once they've moved.

Reply:
I get the transitioning idea. My point is just that we could also start transitioning by routing resources instead of hacking them.

Reply:
The problem is we need parts of the legacy resource. If we can do that via routing resources then I'm fine to switch but I'm not sure how to do that yet.

I like the idea @fgimenez proposed of linking the version bundle to a key helper. That would be better than hacking the `clustertpr` version as we are currently.